docs: update styling support for streaming (#35375)
vercel/next.js: docs/advanced-features/react-18/streaming.md

#### Styling

> So far only `styled-jsx` is supported with streaming. The Next.js team is working on support for CSS modules and other CSS solutions in streaming SSR. Stay tuned for updates.

Reviewer: Can we link to a RFC or Discussion for the updates?

> So far only `styled-jsx` is supported with streaming. The Next.js team is working on support for CSS modules and other CSS solutions in streaming SSR. Stay tuned for updates.

Reviewer 2: We should also mention inlined styles and global solutions (global CSS import, Tailwind CSS) here.

> So far only `styled-jsx` is supported with streaming. The Next.js team is working on support for CSS modules and other CSS solutions in streaming SSR. Stay tuned for updates.

Reviewer 3: `unstable_useFlushEffects` should be sufficient for any library that follows the [recommended upgrade guide](https://github.com/reactwg/react-18/discussions/110). It's just that `styled-jsx` support is built-in.

> React inline styles, Global CSS, CSS modules and Next.js built-in `styled-jsx` are supported with streaming. The Next.js team is working on the guide of integrating other CSS-in-JS solutions in streaming SSR. Stay tuned for updates.

Reviewer 2: ```suggestion
Inline styles, Global CSS, CSS modules and Next.js built-in `styled-jsx` are supported with streaming. The Next.js team is working on the guide of integrating other CSS-in-JS solutions in streaming SSR. Stay tuned for updates.
```

> Notice that for React Server Components, `styled-jsx` should be only placed in client components.

Reviewer: ```suggestion
> Note: `styled-jsx` should be only placed in client components, not server components, when using React Server Components
```
